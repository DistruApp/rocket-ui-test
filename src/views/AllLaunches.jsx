import React  from 'react';
import {useLaunches} from "../hooks/launchesProvider";
import LanchesTable from "../components/LaunchesTable";

const AllLaunchesView = () => {
  const {launches, isfetchingLaunches} = useLaunches();

  if (isfetchingLaunches) {
    return <div> LOADING </div>;
  }

  if (!launches.length) {
    return <div> NO DATA </div>;
  }

  return (
      <div>
          <LanchesTable launches={launches}/>
      </div>
  )
}

export default AllLaunchesView
