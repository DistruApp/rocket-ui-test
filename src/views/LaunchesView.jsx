import React, { useContext } from 'react';
import Launch from '../components/Launch';
import { store } from "../stores/store";

const LaunchesView = () => {
    const {dispatch, state} = useContext(store);

    if (state.error.show) {
        return (
            <div>
                <h2>SpaceX Launches</h2>
                { state.error.msg }
            </div>
        );
    }

    if (state.loading ) {
        return (
            <div>
                <h2>SpaceX Launches</h2>
                <span>
                    Loading launch data...
                </span>
            </div>
        );
    }

    if (state.launches.length > 0) {
        return (
            <div>
                <h2>SpaceX Launches</h2>
                {
                    state.launches.map((launch) => {
                        return (
                            <Launch
                                key={launch.id}
                                launch={launch}
                            />
                        );
                    })
                }
            </div>
        );
    }
    return null;
}

export default LaunchesView;
