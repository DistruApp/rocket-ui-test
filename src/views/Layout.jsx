import React from 'react';
import Navigation from "../components/Navigation";

const Layout = ({ children }) => {

  return (
    <main className={`layout`}>
        <Navigation/>
        <section>
            { children }
        </section>
    </main>
  );
};

export default Layout;
