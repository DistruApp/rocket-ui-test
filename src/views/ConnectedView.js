import React from 'react';
import { connect } from 'react-redux';
import Layout from './Layout';
import Navigation from '../components/Navigation';

const menu = Navigation();

function MasterLayoutHOC(WrappedComponent, pageName) {
	const MasterLayoutImpl = (props) => {
		const layoutProps = {
			menu,
			pageName,
		};

		return (
			<Layout {...layoutProps}>
				<WrappedComponent {...props} />
			</Layout>
		);
	};

	const mapStateToProps = (state) => state;

	const mapDispatchToProps = (dispatch) => ({
		dispatch,
	});

	return connect(mapStateToProps, mapDispatchToProps)(MasterLayoutImpl);
}

export default MasterLayoutHOC;
