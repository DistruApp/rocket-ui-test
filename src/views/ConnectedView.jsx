import React, { Component } from 'react';
import { AnimatorGeneralProvider, Animator } from '@arwes/animation';
import { BleepsProvider } from '@arwes/sounds';
import { ArwesThemeProvider, StylesBaseline } from '@arwes/core';

import Layout from './Layout';
import Navigation from '../components/Navigation';

const ROOT_FONT_FAMILY = '"Titillium Web", sans-serif';
const IMAGE_URL = 'https://playground.arwes.dev/assets/images/wallpaper.jpg';
const SOUND_OBJECT_URL = 'https://playground.arwes.dev/assets/sounds/object.mp3';
const SOUND_TYPE_URL = 'https://playground.arwes.dev/assets/sounds/type.mp3';

const audioSettings = { common: { volume: 0.25 } };

const playersSettings = {
  object: { src: [SOUND_OBJECT_URL] },
  type: { src: [SOUND_TYPE_URL], loop: true }
};

const bleepsSettings = {
  object: { player: 'object' },
  type: { player: 'type' }
};

const generalAnimator = { duration: { enter: 150, exit: 150, stagger: 40 } };

const menu = Navigation();

function MasterLayoutHOC(WrappedComponent, pageName) {
  class MasterLayoutImpl extends Component {
    render() {

      const layoutProps = {
        menu,
        pageName
      };

      return (
        <ArwesThemeProvider>
          <StylesBaseline styles={{ body: { fontFamily: ROOT_FONT_FAMILY } }} />
          <BleepsProvider
            audioSettings={audioSettings}
            playersSettings={playersSettings}
            bleepsSettings={bleepsSettings}
          >
            <AnimatorGeneralProvider animator={generalAnimator}>
              <Layout {...layoutProps}>
                <WrappedComponent {...this.props} />
              </Layout>
            </AnimatorGeneralProvider>
          </BleepsProvider>
        </ArwesThemeProvider>
      );
    }
  }

  return MasterLayoutImpl
}

export default MasterLayoutHOC;
