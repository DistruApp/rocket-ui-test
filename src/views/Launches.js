import React, { useEffect } from 'react';
import ConnectedView from './ConnectedView';
import { fetchLaunchesIfNeeded } from '../actions/Launches';
import Launch from '../components/Launch';

const LaunchesView = (props) => {
	useEffect(() => {
		const { dispatch, launchesCollection } = props;
		fetchLaunchesIfNeeded({ dispatch, launchesCollection });
	}, []);

	const getContent = () => {
		const { launchCollection } = props;

		if (!launchCollection || launchCollection.fetching) {
			return <div> LOADING </div>;
		}

		if (!launchCollection.launches.length) {
			return <div> NO DATA </div>;
		}

		const launches = [];

		for (let i = 0; i < launchCollection.launches.length; i++) {
			const launch = launchCollection.launches[i];

			launches.push(
				<Launch
					{...{
						key: launch.launch_id,
						launch,
					}}
				/>
			);
		}
		return <ul>{launches}</ul>;
	};

	return (
		<div>
			<h2> SpaceX launches </h2>
			{getContent()}
		</div>
	);
};

export default ConnectedView(LaunchesView, 'launches');
