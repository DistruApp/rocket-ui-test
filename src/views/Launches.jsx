import React, { Component } from 'react';
import ConnectedView from './ConnectedView';
import Launch from "../components/Launch";

class LaunchesView extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selectedFlightId: null,
    }
  }

  componentDidMount() {
    const { fetchLanuches } = this.props;
    fetchLanuches();
  }

  getContent() {
    const { spacex } = this.props;
    const { selectedFlightId } = this.state;

    if (spacex.fetchingLaunches) {
      return <div> LOADING </div>;
    }

    if (spacex.launches.length === 0) {
      return <div> NO DATA </div>;
    }

    const launches = Object.values(spacex.launches).map(launch => 
      <Launch {...{
        key: launch.id,
        launch,
        rocketDetails: spacex.rockets[launch.rocket],
        onClickHandler: this.setSelectedFlightId,
        selected: launch.id === selectedFlightId
      }} />
    );

    return launches;
  }

  setSelectedFlightId = (flightNumber = null) => {
    const { spacex, fetchRocket } = this.props
    const { selectedFlightId } = this.state;
    const rocketId = spacex.launches[flightNumber].rocket;
    fetchRocket(rocketId);
    this.setState({ selectedFlightId: selectedFlightId !== flightNumber ? flightNumber : null});
  }

  render() {
    return (
      <div>
        <h2> SpaceX launches </h2>
        {this.getContent()}
      </div>
    );
  }
}

export default ConnectedView(LaunchesView, 'launches');
