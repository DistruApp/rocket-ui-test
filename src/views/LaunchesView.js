import React, { useEffect, useState } from 'react';
import ConnectedView from './ConnectedView';
import { fetchLaunchesIfNeeded } from '../actions/Launches';
import Launch from '../components/Launch';

const LaunchesView = ({ dispatch, launchCollection }) => {
	const [showDetails, setShowDetails] = useState(false);

	const toggleShowDetails = () => {
		setShowDetails((prevState) => !prevState);
	};

	useEffect(() => {
		fetchLaunchesIfNeeded({ dispatch, launchCollection });
	}, []);

	const showAllLaunches = () => {
		const { launches } = launchCollection;

		if (!launchCollection || launchCollection.fetching) {
			return <div> LOADING </div>;
		}

		if (!launchCollection.launches.length) {
			return <div> NO DATA </div>;
		}

		const launchesList = launches.map((launch) => {
			return (
				<Launch
					key={launch.id}
					launch={launch}
					showDetails={showDetails}
					toggleShowDetails={toggleShowDetails}
				/>
			);
		});

		return <ul>{launchesList}</ul>;
	};

	return (
		<div>
			<h2> SpaceX launches </h2>
			{showAllLaunches()}
		</div>
	);
};

export default ConnectedView(LaunchesView, 'launches');
