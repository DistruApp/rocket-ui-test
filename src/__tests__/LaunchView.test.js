import React from 'react';
import { configure, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import * as redux from "react-redux";
import { LaunchesView } from '../views/LaunchesView';

const mockDispatch = jest.fn();
jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
  useDispatch: () => mockDispatch
}));

const exampleLaunchDef = {
  rocket:{
    rocket_id: '1'
  },
  _id: '1',
  mission_name: 'Test Mission Name',
  flight_number: 'Test Flight Number'
}

configure({ adapter: new Adapter() });

describe('LaunchView component test', () => {
  it('renders LaunchView component without crashing in Error', () => {
    const spy = jest.spyOn(redux, 'useSelector')
    spy.mockReturnValue({ fetching: false, launches: 1 })
    const component = mount(<LaunchesView/>);
    expect(component.getElements()).toMatchSnapshot();
  });
  it('renders LaunchView component without crashing', () => {
    const spy = jest.spyOn(redux, 'useSelector')
    spy.mockReturnValue({ fetching: false, launches: [exampleLaunchDef] })
    const component = mount(<LaunchesView/>);
    expect(component.getElements()).toMatchSnapshot();
  });
});