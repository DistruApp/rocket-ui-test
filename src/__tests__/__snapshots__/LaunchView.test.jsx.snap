// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LaunchView component test renders LaunchView component without crashing 1`] = `
Array [
  <LaunchesView />,
]
`;

exports[`LaunchView component test renders LaunchView component without crashing in Error 1`] = `
Array [
  <LaunchesView />,
]
`;
